File: javascript/src/models/event_type_from_open_api.ts
// this file is @generated

export interface EventTypeFromOpenApi {
  deprecated: boolean;
  description: string;
  featureFlag?: string | null;
  /** The event type group's name */
  groupName?: string | null;
  /** The event type's name */
  name: string;
  schemas?: any | null;
}

export namespace EventTypeFromOpenApiUtil {
  export function _fromJsonObject(object: any): EventTypeFromOpenApi {
    return {
      deprecated: object["deprecated"],
      description: object["description"],
      featureFlag: object["featureFlag"],
      groupName: object["groupName"],
      name: object["name"],
      schemas: object["schemas"],
    };
  }

  export function _toJsonObject(self: EventTypeFromOpenApi): any {
    return {
      deprecated: self.deprecated,
      description: self.description,
      featureFlag: self.featureFlag,
      groupName: self.groupName,
      name: self.name,
      schemas: self.schemas,
    };
  }
}

File: kotlin/lib/src/main/kotlin/models/EventTypeFromOpenApi.kt
// This file is @generated
package com.svix.kotlin.models

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonObject

@Serializable
data class EventTypeFromOpenApi(
    val deprecated: Boolean,
    val description: String,
    val featureFlag: String? = null,
    val groupName: String? = null,
    val name: String,
    val schemas: JsonObject? = null,
)
